transcendence-app/src/game/infrastructure/db/postgres/game.postgres.repository.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { BasePostgresRepository } from '../../../../shared/db/postgres/postgres.repository';
+import { table } from '../../../../shared/db/models';
+import { Game, gameKeys } from '../../db/game.entity';
+import { IGameRepository } from '../game.repository';
+import { PostgresPool } from '../../../../shared/db/postgres/postgresConnection.provider';
+import { makeQuery } from '../../../../shared/db/postgres/utils';
+import { PaginationWithSearchQueryDto } from '../../../../../src/shared/dtos/pagination-with-search.query.dto';
+
+@Injectable()
+export class GamePostgresRepository
+  extends BasePostgresRepository<Game>
+  implements IGameRepository
+{
+  constructor(protected pool: PostgresPool) {
+    super(pool, table.GAME, Game);
+  }
+
+  addGame(game: Game): Promise<Game | null> {
+    return this.add(game);
+  }
+
+  async getById(id: string): Promise<Game | null> {
+    const games = await this.getByKey(gameKeys.ID, id);
+    return games && games.length ? games[0] : null;
+  }
+
+  async deleteById(id: string): Promise<Game | null> {
+    const games = await this.deleteByKey(gameKeys.ID, id);
+    return games && games.length ? games[0] : null;
+  }
+
+  async getPaginatedGames(
+    paginationDto: Required<PaginationWithSearchQueryDto>,
+  ): Promise<Game[] | null> {
+    const { limit, offset, sort, search } = paginationDto;
+    const gamesData = await makeQuery<Game>(this.pool, {
+      text: `SELECT *
+      FROM ${this.table}
+      WHERE ${gameKeys.PLAYERONEUSERNAME} ILIKE $1
+      OR ${gameKeys.PLAYERTWOUSERNAME} ILIKE $1
+      ORDER BY ${sort}
+      LIMIT $2
+      OFFSET $3;`,
+      values: [`%${search}%`, limit, offset],
```

Comment:
and remove these percents % here.

Reply:
These two comments for removing the I from iLike and removing the percents (%) were only valid to retrieve games for a particular user. If we're retrieving the games for any user matching the search, then we still need the i in iLike and the two percents (%). Think about what returns what, and stay consistent.